solidity/contracts/bridge/Bridge.sol

```diff
@@ -1339,6 +1406,155 @@ contract Bridge is Ownable, EcdsaWalletOwner {
+            "Wallet is neither in Live nor MovingFunds state"
+        );
+
+        frauds.submitFraudChallenge(walletPublicKey, sighash, signature);
+    }
+
+    /// @notice Allows to defeat a pending fraud challenge against a wallet if
+    ///         the transaction that spends the UTXO follows the protocol rules.
+    ///         In order to defeat the challenge the same `walletPublicKey` and
+    ///         signature (represented by `r`, `s` and `v`) must be provided as
+    ///         were used in the fraud challenge. Additionally a preimage must
+    ///         be provided which was used to calculate the sighash during input
+    ///         signing. The fraud challenge defeat attempt will only succeed if
+    ///         the inputs in the preimage are considered honestly spent by the
+    ///         wallet. Therefore the transaction spending the UTXO must be
+    ///         proven in the Bridge before a challenge defeat is called.
+    ///         If successfully defeated, the fraud challenge is closed and the
+    ///         amount of ether deposited by the challenger is sent to the
+    ///         treasury.
+    /// @param walletPublicKey The public key of the wallet in the uncompressed
+    ///        and unprefixed format (64 bytes).
+    /// @param preimage The preimage which produces sighash used to generate the
+    ///        ECDSA signature that is the subject of the fraud claim. It is a
+    ///        serialized subset of the transaction. The exact subset used as
+    ///        the preimage depends on the transaction input the signature is
+    ///        produced for. See BIP-143 for reference.
+    /// @param signature Bitcoin signature in the R/S/V format.
+    /// @param witness Flag indicating whether the preimage was produced for a
+    ///        witness input. True for witness, false for non-witness input.
+    /// @dev Requirements:
+    ///      - `walletPublicKey`, signature (represented by `r`, `s` and `v`) as
+    ///        well `sighash` calculated as hash256(preimage) must identify an
+    ///        open fraud challenge
+    ///      - the preimage must be generated according to the protocol
+    ///      - before a defeat attempt is made the transaction that spends the
+    ///        given UTXO must be proven in the Bridge.
+    function defeatFraudChallenge(
+        bytes memory walletPublicKey,
+        bytes memory preimage,
+        Frauds.RSVSignature calldata signature,
+        bool witness
+    ) external {
+        uint256 utxoKey = frauds.unwrapChallenge(
+            walletPublicKey,
+            preimage,
+            signature,
+            witness
+        );
+
+        // Check that the UTXO key identifies a correctly spent UTXO.
+        require(
+            deposits[utxoKey].sweptAt > 0 || spentMainUTXOs[utxoKey],
+            "Spent UTXO not found among correctly spent UTXOs"
+        );
+
+        frauds.finalizeFraudChallengeDefeat(
```

Comment:
Maybe just `frauds.defeatChallenge`? It is kind of related to the discussion we had with @lukasz-zimnoch [here](https://github.com/keep-network/tbtc-v2/pull/177#discussion_r834107047).

Reply:
Done in 9c85e4483857e9487c8d7bddcaf9bd2df792dfdd.